localization_manager/src/localization_manager_node.cpp

```diff
@@ -0,0 +1,172 @@
+ * the License.
+ */
+#include "localization_manager/localization_manager_node.hpp"
+#include <boost/bind.hpp>
+#include <boost/bind/placeholders.hpp>
+#include <carma_ros2_utils/timers/ROSTimerFactory.hpp>
+
+namespace localization_manager
+{
+    namespace std_ph = std::placeholders;
+
+    Node::Node(const rclcpp::NodeOptions &options)
+        :carma_ros2_utils::CarmaLifecycleNode(options)
+
+    {
+        //Create initial config
+        config_ = LocalizationManagerConfig();
+
+        //Declare parameters
+        config_.fitness_score_degraded_threshold = declare_parameter<double>("fitness_score_degraded_threshold", config_.fitness_score_degraded_threshold);
+        config_.fitness_score_fault_threshold = declare_parameter<double>("fitness_score_fault_threshold",config_.fitness_score_fault_threshold);
+        config_.ndt_frequency_degraded_threshold = declare_parameter<double>("ndt_frequency_degraded_threshold", config_.ndt_frequency_degraded_threshold);
+        config_.ndt_frequency_fault_threshold = declare_parameter<double>("ndt_frequency_fault_threshold", config_.ndt_frequency_fault_threshold);
+        config_.auto_initialization_timeout = declare_parameter<int>("auto_initialization_timeout", config_.auto_initialization_timeout);
+        config_.gnss_only_operation_timeout = declare_parameter<int>("gnss_only_operation_timeout", config_.gnss_only_operation_timeout);
+        config_.sequential_timesteps_until_gps_operation = declare_parameter<int>("sequential_timesteps_until_gps_operation", config_.sequential_timesteps_until_gps_operation);
+        config_.gnss_data_timeout = declare_parameter<int>("gnss_data_timeout", config_.gnss_data_timeout);
+        config_.localization_mode = declare_parameter<int>("localization_mode", config_.localization_mode);
+        config_.pose_pub_rate = declare_parameter<double>("pose_pub_rate", config_.pose_pub_rate);
+
+    }
+
+    rcl_interfaces::msg::SetParametersResult Node::parameter_update_callback(const std::vector<rclcpp::Parameter> &parameters)
+    {
+        auto error = update_params<double>({{"fitness_score_degraded_threshold", config_.fitness_score_degraded_threshold},
+        {"fitness_score_fault_threshold",config_.fitness_score_fault_threshold},
+        {"ndt_frequency_degraded_threshold", config_.ndt_frequency_degraded_threshold},
+        {"ndt_frequency_fault_threshold", config_.ndt_frequency_fault_threshold},
+        {"pose_pub_rate", config_.pose_pub_rate},
+        },parameters);
+
+        auto error_2 = update_params<int>({{"auto_initialization_timeout", config_.auto_initialization_timeout},
+        {"gnss_only_operation_timeout", config_.gnss_only_operation_timeout},
+        {"sequential_timesteps_until_gps_operation", config_.sequential_timesteps_until_gps_operation},
+        {"gnss_data_timeout", config_.gnss_data_timeout},
+        {"localization_mode", config_.localization_mode}
+        }, parameters);
```

Comment:
Since none of these new values are actually passed to the worked this function is meaningless. For this particular node, I would be ok not having runtime configurability because of the safety risk it poses to change localization modes mid run.